app/models/podcast.rb

```diff
@@ -4,4 +4,12 @@ class Podcast < ApplicationRecord
 
   validates :title, presence: true
   validates :title, length: { minimum: 2 }
+
+  def self.random_by_day
+    podcast_count = count
+    return if podcast_count.zero?
+    new_random_podcast = offset(rand(podcast_count)).first
+    cached_random_podcast_id = Rails.cache.fetch("random_podcast_id", expires_in: 1.days) { new_random_podcast.id }
+    find_by(id: cached_random_podcast_id)
+  end
```

Comment:
1) you need to stylize code like:

```ruby
def foo
  return if false
  # placing newline
  ...
end
```

2) Is more cheapest query to the database for checking for existing records is `SELECT 1 FROM podcasts`, but your solution uses `.count` which makes `SELECT COUNT(*) FROM podcasts;`. Instead of `.count` pretty sure using `.exists?`

3) `new_random_podcast ` is always executing sql query and this is a mistake because every time make query to the database. In the issue #5 ive says why we caching every_day_podcast - because they querying to the database every time. Now we got same result. Place it right to the block at Rails.cache.fetch and its stopped to querying every time